### DISCUSSION TITLE
How to get fine grained auth0 permission control?

### DISCUSSION THREAD

**Discussion by RomuloPBenedetti** [GitbookIO/community#748] [1. How to] [answered]
How can we control what pages can be accessed with auth0? Do the integration takes roles in consideration?

**Comment by SamyPesse** [comment, marked as answer]
This is not yet supported. Currently the auth0 integration act as an authentication layer on the entire content. But we plan on supporting fine-grained permission control at the page level.

**Reply by RomuloPBenedetti** [reply]
Thanks! Good to know.

We stumbled on the need to only make half of an API visible, and it is not possible on the same site. Even with different sites, roles would be a better solution. At the moment, we create apps for each site in Auth0 and restrict through post-login actions + user metadata.

Would Also make easier, to create a more solid login/logout experience.
